examples/passing/3187-UnusedNameClash.purs

```diff
@@ -0,0 +1,12 @@
+module Main (main) where
+
+import Prelude ((+))
+import Control.Monad.Eff.Console (log)
+
+-- the __unused parameter used to get optimized away
+abuseUnused :: forall a. a -> a
+abuseUnused __unused = __unused
```

Comment:
I don't understand why this is inside `passing`, shouldn't this explode now that `UnusedIdent` is a special, generated name inside the compiler?

Reply:
No, the opposite. It exploded before this PR, because `"__unused"` was a magic string. This PR makes it so `__unused` is no longer a magic name, but a name like any other. So this test now passes.

Reply:
Oh I see :D I somehow thought this was a self recursive ever looping binding. My bad.